### DISCUSSION TITLE
Show: Blog theme I'm developing, "Eleventy Satisfactory"

### DISCUSSION THREAD

**Discussion by mendhak** [11ty/eleventy#2778] [Show and tell]
As I've been asking many questions in here recently I thought I'd share the [blog theme I've been working on](https://code.mendhak.com/eleventy-satisfactory/).  Thanks to you all answering here as well as many people who asked  questions in the past, I found searching old issues and discussions very valuable! 

Since I'm migrating from another theme, I worked on various convenience functions for the blog to help me out.  

A [cssminification shortcode](https://github.com/mendhak/eleventy-satisfactory/blob/main/_includes/layouts/base.njk#L24-L31) that combines multiple files.  

My favorite is a [Github repo card](https://code.mendhak.com/eleventy-satisfactory/github-repo-card/).   

Another favorite is a bit more low-key, being able to [link to another post using its `.md` file name](https://code.mendhak.com/eleventy-satisfactory/posting-links/), and it gets [converted to its final URL](https://github.com/mendhak/eleventy-satisfactory/blob/main/_configs/markdownlibrary.renderer.links.js).  Sadly I couldn't figure out how to make this work inside paired shortcodes though.  :disappointed: 

Converting [Markdown images to use a lightbox and also a figure shortcode with width options](https://code.mendhak.com/eleventy-satisfactory/post-with-an-image/).  

A [gallery shortcode that just takes normal MD images](https://code.mendhak.com/eleventy-satisfactory/post-with-a-gallery/) too.  

[Notice boxes](https://code.mendhak.com/eleventy-satisfactory/post-notice/). 

Doing this I came to appreciate how flexible 11ty is, its default behaviors are good enough, and it can be built upon or extended quite easily.  I still haven't gotten my head wrapped properly around some of its concepts, I think I've been operating at a conceptual level on some of them.  But it's all good.

**Comment by cfjedimaster** [comment]
I like this. How would you handle additional navigation, ie if you wanted some links on top, with the bottom links for being less important stuff?

**Reply by mendhak** [reply]
Thanks for that.  I've deliberately removed the top navigation links as I didn't want any kind of navigation present.  It does seem straightforward to add back if you need though, the base I generated from has it:  https://github.com/lkhrs/eleventy-simple/blob/main/_includes/layouts/base.njk#L25-L28
